Build on macos M1 / Apple Silicon fails (GL/mobile-related)

I encountered this error when setting up a project for use in the tutorial:
```
go run ./cmd/tut/
```
```
# golang.org/x/mobile/gl
In file included from /Users/oubiwann/go/pkg/mod/golang.org/x/mobile@v0.0.0-20200329125638-4c31acba0007/gl/work.go:29:
./work.h:18:10: fatal error: 'OpenGLES/ES2/glext.h' file not found
#include <OpenGLES/ES2/glext.h>
```

I tried with:
* Go versions 1.16, 1.17
* engo version 1.0.4, 1.0.5, 1.0.6

After finding [this bug](https://github.com/golang/go/issues/43237), and [the latest version of ](https://pkg.go.dev/golang.org/x/mobile?tab=versions), I manually set this in `go.mod`:
```
	golang.org/x/mobile v0.0.0-20211109191125-d61a72f26a1a
```

and got past that error. I now see the following errors when using the latest `x/mobile` in all combinations of the versions listed above:
```
# github.com/EngoEngine/gl
/Users/oubiwann/go/pkg/mod/github.com/!engo!engine/gl@v1.0.12/gl_mobile.go:437:17: cannot refer to unexported name gl.line
/Users/oubiwann/go/pkg/mod/github.com/!engo!engine/gl@v1.0.12/gl_mobile.go:437:24: syntax error: unexpected newline, expecting comma or }
# golang.org/x/mobile/app
darwin_desktop.m:51:1: warning: method possibly missing a [super prepareOpenGL] call [-Wobjc-missing-super-calls]
```

Thoughts? Work-arounds?

--- a/engo_glfw.go
+++ b/engo_glfw.go
@@ -1,5 +1,11 @@
-// +build darwin,!arm,!arm64 linux windows
-// +build !ios,!android,!js,!sdl,!headless,!vulkan
+//go:build (darwin || linux || windows) && !ios && !android && !js && !sdl && !headless && !vulkan
+// +build darwin linux windows
+// +build !ios
+// +build !android
+// +build !js
+// +build !sdl
+// +build !headless
+// +build !vulkan
 
 package engo
 

--- a/engo_mobile.go
+++ b/engo_mobile.go
@@ -1,5 +1,6 @@
-//+build android darwin,arm darwin,arm64 ios
-//+build !mobilebind
+//go:build (android || ios) && !mobilebind
+// +build android ios
+// +build !mobilebind
 
 package engo
 

--- a/engo_mobile_bind_ios.go
+++ b/engo_mobile_bind_ios.go
@@ -1,4 +1,4 @@
-//+build ios darwin,arm darwin,arm64
+//+build ios
 //+build mobilebind
 
 package engo

--- a/gamepads_glfw.go
+++ b/gamepads_glfw.go
@@ -1,5 +1,11 @@
-// +build darwin,!arm,!arm64 linux windows
-// +build !ios,!android,!js,!sdl,!headless,!vulkan
+//go:build (darwin || linux || windows) && !ios && !android && !js && !sdl && !headless && !vulkan
+// +build darwin linux windows
+// +build !ios
+// +build !android
+// +build !js
+// +build !sdl
+// +build !headless
+// +build !vulkan
 
 package engo
 

--- a/go.mod
+++ b/go.mod
@@ -2,13 +2,13 @@ module github.com/EngoEngine/engo
 
 require (
 	github.com/EngoEngine/ecs v1.0.5
-	github.com/EngoEngine/gl v1.0.12
+	github.com/EngoEngine/gl v1.0.13
 	github.com/EngoEngine/math v1.0.4
 	github.com/Noofbiz/sdlMojaveFix v0.0.1
 	github.com/Noofbiz/tmx v0.2.0
 	github.com/go-bindata/go-bindata v3.1.2+incompatible
-	github.com/go-gl/gl v0.0.0-20210501111010-69f74958bac0 // indirect
-	github.com/go-gl/glfw/v3.3/glfw v0.0.0-20210410170116-ea3d685f79fb
+	github.com/go-gl/gl v0.0.0-20211210172815-726fda9656d6 // indirect
+	github.com/go-gl/glfw/v3.3/glfw v0.0.0-20211213063430-748e38ca8aec
 	github.com/go-gl/mathgl v1.0.0
 	github.com/golang/freetype v0.0.0-20170609003504-e2365dfdc4a0
 	github.com/hajimehoshi/go-mp3 v0.3.2
@@ -19,14 +19,14 @@ require (
 	github.com/srwiley/oksvg v0.0.0-20210519022825-9fc0c575d5fe
 	github.com/srwiley/rasterx v0.0.0-20210519020934-456a8d69b780
 	github.com/stretchr/testify v1.7.0
-	github.com/veandco/go-sdl2 v0.4.8
+	github.com/veandco/go-sdl2 v0.4.14
 	github.com/vulkan-go/glfw v0.0.0-20210402172934-58379a80228d
 	github.com/vulkan-go/vulkan v0.0.0-20210402152248-956e3850d8f9
 	golang.org/x/exp v0.0.0-20210618142145-ffcf9a09ea36 // indirect
-	golang.org/x/image v0.0.0-20210607152325-775e3b0c77b9
-	golang.org/x/mobile v0.0.0-20210614202936-7c8f154d1008
-	golang.org/x/net v0.0.0-20210614182718-04defd469f4e // indirect
-	golang.org/x/sys v0.0.0-20210616094352-59db8d763f22 // indirect
+	golang.org/x/exp/shiny v0.0.0-20220218215828-6cf2b201936e // indirect
+	golang.org/x/image v0.0.0-20211028202545-6944b10bf410
+	golang.org/x/mobile v0.0.0-20220224134551-8a0a1e50732f
+	golang.org/x/sys v0.0.0-20220224120231-95c6836cb0e7 // indirect
 	gopkg.in/check.v1 v1.0.0-20190902080502-41f04d3bba15 // indirect
 	gopkg.in/yaml.v2 v2.2.8 // indirect
 )

--- a/input_keys.go
+++ b/input_keys.go
@@ -1,4 +1,5 @@
-//+build js android ios darwin,arm darwin,arm64 headless
+//go:build js || android || ios || headless
+// +build js android ios headless
 
 package engo
 

--- a/input_keys_glfw.go
+++ b/input_keys_glfw.go
@@ -1,5 +1,11 @@
-// +build darwin,!arm,!arm64 linux windows
-// +build !ios,!android,!js,!sdl,!headless,!vulkan
+//go:build (darwin || linux || windows) && !ios && !android && !js && !sdl && !headless && !vulkan
+// +build darwin linux windows
+// +build !ios
+// +build !android
+// +build !js
+// +build !sdl
+// +build !headless
+// +build !vulkan
 
 package engo
 

